app/hooks/use-display-currency.ts

```diff
@@ -60,34 +69,73 @@ export const useDisplayCurrency = () => {
     }).format(amount)
   }, [])
 
-  const fiatSymbol = useMemo(
-    () => currencyList.find((currency) => currency.id === displayCurrency)?.symbol ?? "$",
-    [currencyList, displayCurrency],
-  )
-
   // FIXME this should come from the backend and should be used in currency inputs
   const minorUnitToMajorUnitOffset = 2
 
-  const moneyAmountToMajorUnitOrSats = (
-    moneyAmount: MoneyAmount<WalletOrDisplayCurrency>,
-  ) => {
-    return moneyAmount.currency === WalletCurrency.Btc
-      ? moneyAmount.amount
-      : moneyAmount.amount / 10 ** minorUnitToMajorUnitOffset
-  }
+  const moneyAmountToMajorUnitOrSats = useCallback(
+    (moneyAmount: MoneyAmount<WalletOrDisplayCurrency>) => {
+      switch (moneyAmount.currency) {
+        case WalletCurrency.Btc:
+          return moneyAmount.amount
+        case WalletCurrency.Usd:
+          return moneyAmount.amount / 100
+        case DisplayCurrency:
+          return moneyAmount.amount / 10 ** minorUnitToMajorUnitOffset
+      }
+    },
+    [minorUnitToMajorUnitOffset],
+  )
 
-  const moneyAmountToTextWithUnits = useCallback(
+  const amountInMajorUnitOrSatsToMoneyAmount = useCallback(
+    (
+      amount: number,
+      currency: WalletOrDisplayCurrency,
+    ): MoneyAmount<WalletOrDisplayCurrency> => {
```

Comment:
in the backend we have a rule to use an object argument when there is more than one input, ie

```suggestion
    ({amount, currency}: {
      amount: number,
      currency: WalletOrDisplayCurrency,
    }): MoneyAmount<WalletOrDisplayCurrency> => {
```

Reply:
do we want to enforce this rule here? I think that make sense to have the same rules to have less context switch when working on both code base simultaneously

Reply:
Yeah I think it makes sense to use that rule